### ISSUE TITLE
set the cores limit on cluster execution 

### ISSUE THREAD

**Issue by shizipo** [snakemake/snakemake#465]
here is my command : 
snakemake -s Snakefile_S1.py --cluster "sbatch -N 1 -n {threads} -p node2 --mem-per-cpu=4000"  --latency-wait 60  --cores 20

I want to limit the cores 150 CPUs for my Snakefile_S1.py ( my host machine has 200 CPUs) . 

I set --cores 20 ，just use 20*2 ( 40CPUs )，the  --cores equal to --jobs ??

I also use --local-cores 150 ，-j 20 ， there just run 20 jobs .

i do know what did I do wrong , can you tell me how to set it ?

**Comment by SultanOrazbayev** [comment]
Local cores refer to the cores on the machine from which snakemake workflow is started. If you are on the cluster this will typically be just the login node, so setting `local-cores` to 150 is not advisable (depending on the cluster configuration, your job will probably be cancelled right away).

For your purpose, it's advisable to use `--jobs 1` and `--cores 150` (or if you want to have 10 concurrent jobs with 15 cores each, then you can adjust the numbers accordingly).

Your final command will probably look like:
`snakemake -s Snakefile_S1.py -j 1 --latency-wait 60  --cluster "sbatch -N 1 -n {threads} -p node2 --cores 20 --mem-per-cpu=4000"`

**Comment by SultanOrazbayev** [comment]
Note that the `--cores` argument is passed to the cluster.

**Comment by shizipo** [comment]
snakemake -np -s Snakefile_S1.py 
Job counts:
        count   jobs
        359     CallDuplexConsensusReads_06bam
        367     ClipBam
        359     CollectDuplexSeqMetrics
        283     ExtractUmisFromBam
        280     FastqToSam
        367     FilterConsensusReads
        292     GroupReadsByUmi
        290     MergeBamAlignment
        367     MergeBamAlignment2                 #2 cores 
        288     SamToFastq                                   #2 cores
        361     SamToFastq2                                 #2 cores
        760     bwa                                               #15 cores
        1       all

i have many steps in  Snakefile_S1.py  and i set differents threads in each steps  , so i use "-n {threads}".   i want to set a limit of total cores  one time rather than jobs limit 。

**Comment by VGalata** [comment]
After reading [issue 213](https://github.com/snakemake/snakemake/issues/213), I have the impression that you cannot limit the number of cores used by `snakemake` in cluster mode: using `--cores` does not affect the number of cores for the submitted jobs. You can use `--jobs` to limit the number of concurrent jobs but the number of cores to be used comes from the cluster call.

**Comment by SichongP** [comment]
I don't think you can limit total threads in a cluster mode but there is a way to get around it:  You can use `resources` to limit number of total cores snakemake is allowed to use to schedule jobs. And in your rules you just need to put threads in `resources` directive instead.
